### ISSUE TITLE
talkback not announcing header accessibilityRole

### ISSUE THREAD

**Issue by tburra** [facebook/react-native#32448]
### Description

Talkback is not announcing the header role after reading the text. It just reads the text and does not read 'header' after the text. Other accessibility roles work, header role is broken. 
<Text accessibilityRole={"header"} 

I see that the support was added in this pull request but it isn't working currently https://github.com/facebook/react-native/pull/22447

This issue was also mentioned in the comments on https://github.com/facebook/react-native/issues/29457

### Version

0.64.2

### Output of `react-native info`

info Fetching system and libraries information...
System:
    OS: macOS 11.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 19.73 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.21.0 - /var/folders/73/rc4dqz6n5ln9089d0s9mbzj80000gn/T/yarn--1634770418397-0.43470384820320307/node
    Yarn: 1.22.10 - /var/folders/73/rc4dqz6n5ln9089d0s9mbzj80000gn/T/yarn--1634770418397-0.43470384820320307/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v12.21.0/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.2 => 0.64.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
✨  Done in 9.73s.

### Steps to reproduce

Add some text with accessibilityRole header
<Text accessibilityRole={"header"} 

It will read the text, but the suffix header does not get read on talkback
